### PR TITLE
perf(core): StringId hold bytes to avoid decode/encode

### PR DESCRIPTION
**Pull request by javeme**
Change-Id: I1ff93a2ee31f142fb7ab5cfe0e038ef2da2f982d

### CODE REVIEW

File: hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/backend/id/IdGenerator.java [hunk 1]
```diff
@@ -128,14 +129,19 @@ public static int compareType(Id id1, Id id2) {
     public static class StringId implements Id {
 
         protected String id;
+        protected byte[] bytes;
```
Comment (imbajin) [inline hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/backend/id/IdGenerator.java:132]: **Thread Safety Concern**: The lazy initialization pattern used here is not thread-safe. If multiple threads access the same StringId instance simultaneously, race conditions could occur during the lazy initialization of `id` and `bytes` fields. Consider using volatile fields or synchronization to ensure thread safety.

File: hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/backend/id/IdGenerator.java [hunk 2]
```diff
@@ -128,14 +129,19 @@ public static int compareType(Id id1, Id id2) {
...
 
         public StringId(String id) {
```
Comment (imbajin) [inline hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/backend/id/IdGenerator.java:134]: **Inconsistent State Risk**: The constructor initializes `bytes = null` explicitly, but the state where both `id` and `bytes` are null is not handled. Consider adding validation to ensure at least one field is non-null, or document the invariant that exactly one field should be null at any time.

File: hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/backend/id/IdGenerator.java [hunk 3]
```diff
@@ -174,25 +186,34 @@ public int compareTo(Id other) {
             if (cmp != 0) {
                 return cmp;
             }
-            return this.id.compareTo(other.asString());
+            if (this.id != null) {
+                return this.id.compareTo(other.asString());
+            } else {
+                return Bytes.compare(this.bytes, other.asBytes());
```
Comment (imbajin) [inline hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/backend/id/IdGenerator.java:195]: **Performance Optimization**: Consider using `Arrays.equals()` instead of `Bytes.equals()` for byte array comparison if `Bytes.equals()` is just a wrapper. Also, cache the result of `other.asBytes()` to avoid multiple conversions in the comparison.

File: hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/backend/id/IdGenerator.java [hunk 4]
```diff
@@ -174,25 +186,34 @@ public int compareTo(Id other) {
...
-            if (!(other instanceof StringId)) {
+        public boolean equals(Object obj) {
+            if (!(obj instanceof StringId)) {
                 return false;
             }
-            return this.id.equals(((StringId) other).id);
+            StringId other = (StringId) obj;
+            if (this.id != null) {
+                return this.id.equals(other.asString());
+            } else {
```
Comment (imbajin) [inline hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/backend/id/IdGenerator.java:214]: **Potential Memory Leak**: In the equals method, calling `other.asBytes()` when `this.id != null` may trigger unnecessary byte array creation and caching in the other object. Consider checking if the other object has the same representation type first to avoid forced conversions.

File: hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/backend/id/IdGenerator.java [hunk 5]
```diff
@@ -128,14 +129,19 @@ public static int compareType(Id id1, Id id2) {
     public static class StringId implements Id {
```
Comment (imbajin) [inline hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/backend/id/IdGenerator.java:129]: **Missing Test Coverage**: This performance optimization changes core behavior of StringId. Please ensure comprehensive test coverage including:\n- Thread safety tests with concurrent access\n- Edge cases with empty/null inputs\n- Performance regression tests\n- Memory usage validation tests

File: hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/memory/consumer/impl/id/StringIdOffHeap.java [hunk 6]
```diff
@@ -70,6 +70,7 @@ public void serializeSelfToByteBuf(MemoryPool memoryPool) {
     @Override
     public void releaseOriginalVarsOnHeap() {
         this.id = null;
+        this.bytes = null;
```
Comment (javeme) [inline hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/memory/consumer/impl/id/StringIdOffHeap.java:78]: in serializeSelfToByteBuf we can set stringBytes = this.bytes if this.bytes != null

File: hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/backend/id/IdGenerator.java (hunk 1 above)
Comment (javeme) [reply hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/backend/id/IdGenerator.java:132]: At present, I haven't found any scenarios where this would cause problems. It's safe even if multiple assignments occur at the same time.

File: hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/backend/id/IdGenerator.java (hunk 2 above)
Comment (javeme) [reply hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/backend/id/IdGenerator.java:134]: id is null is not allowed in this constructor

File: hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/backend/id/IdGenerator.java (hunk 4 above)
Comment (javeme) [reply hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/backend/id/IdGenerator.java:214]: done

**Review by VGalaxies — APPROVED** [review]
